src/dual.jl

```diff
@@ -387,8 +387,6 @@ Base.AbstractFloat(d::Dual{T,V,N}) where {T,V,N} = convert(Dual{T,promote_type(V
 # General Mathematical Operations #
 ###################################
 
-@inline Base.conj(d::Dual) = d
```

Comment:
If we don't want to drop all the previous DiffRules compact, a `hasmethod` check could be used instead?

Reply:
Do we care about maintaining compatibility with old versions of DiffRules? My personal preference is to always drop old versions as it means less maintenance headaches.

Reply:
If it is so easy, like in this case, might as well, or?

Reply:
Fine, can you add the change as  a suggestion? (I'm not 100% sure how to do it in a way that's precompile-compatible.)

Reply:
Should just be

```jl
if !hasmethod(conj, Tuple{ForwardDiff.Dual})
    @inline Base.conj(d::Dual) = d
end
```

Not sure how to add that as a suggestion.

Reply:
Use "```suggestion"

```suggestion
if !hasmethod(conj, Tuple{ForwardDiff.Dual})
    @inline Base.conj(d::Dual) = d
end
```

Reply:
Is `hasmethod` going to work here? The method you are checking for is defined in just below this in the file.

Reply:
And, even with this mechanism, I think it needs new bounds in Project.toml before tagging. 

In the registry all past versions are marked incompatible with DiffRules 1.1 & 1.2, see https://github.com/JuliaRegistries/General/pull/40252 and friends. The next tagged version should probably accept `<= 1.02 + >= 1.21`.

Reply:
I'm not sure what you are suggestion. Perhaps make a ```suggestion ?

Reply:
I made a PR, weeks ago. But read my comments carefully, there are two concerns. One is whether this `hasmethod` actually works as claimed, did you test it with DiffRules 0.9 say? The other is that tagging this would re-declare compatibility of this package with versions of DiffRules which contain 3-ary rules, which cause it to crash, #530.

Reply:
Let's go with the compat restricted version then. I don't think it will be any big problem in practice.